cmd/all-in-one/main.go

```diff
@@ -302,7 +302,7 @@ func createMetricsQueryService(
 	logger *zap.Logger,
 	metricsReaderMetricsFactory metrics.Factory,
 ) (querysvc.MetricsQueryService, error) {
-	if err := metricsReaderFactory.Initialize(logger); err != nil {
+	if err := metricsReaderFactory.InitializeMetricsFactory(logger); err != nil {
```

Comment:
please revert this change. We have a pattern of factories having `Initialize` method, it needs to stay consistent. And calling it `metricsFactory.InitializeMetricsFactory` is just redundant, only makes way more changes to the code in the PR without adding anything.

Reply:
The reason I made this change is because there are instances where both the `Factory` and `MetricsFactory` interface are used. Both these interfaces have `Initialize` methods but with different parameters. So any type trying to implement both the interfaces gets confused...e.g: `fakeStorageExt` in `cmd/jaeger/internal/extension/jaegerquey/server_test.go`.

Reply:
Understood. I will revert this change.